### PR TITLE
<d2l-input>

### PR DESCRIPTION
**Pull request by vsumner**
I've refactored this to use `iron-input`. 
- [ ] Update docs with use information
- [ ] Update demo with more examples

### CODE REVIEW

File: d2l-input.html [hunk 1]
```diff
@@ -11,39 +12,298 @@
...
 	<template>
     <style include="d2l-input-shared-styles">
-		:host {
+		:host > input {
 			  @apply(--d2l-input);
 			  @apply(--d2l-input-with-text);
 		  }
 
-		  :host(:[aria-invalid='true']) {
+		  :host > input:[aria-invalid='true'] {
```
Comment (dlockhart) [inline d2l-input.html]: Should this be `:host > input[aria-invalid="true"] {`? (no colon after `input`)

File: d2l-input.html [hunk 2]
```diff
@@ -11,39 +12,298 @@
...
+					type: String
+				},
+
+				/**
+				 * The value for this input.
+				 * @type String
+				 */
+				value: {
+					type: String,
+					notify: true
```
Comment (dlockhart) [inline d2l-input.html]: Is `notify` needed here? If so, why not on any of the other properties?

File: d2l-input.html (hunk 1 above)
Comment (vsumner) [inline d2l-input.html]: doh. missed that from the last PR

File: d2l-input.html (hunk 2 above)
Comment (vsumner) [inline d2l-input.html]: Yes it is in the requirements from `iron-input` to work with `bind-value`
```
bind this to the `<input is="iron-input">`'s `bindValue` property, or the value property of your input that is `notify:true`.
```

**Pull request closed without merging**